LDAR_Sim/model_code/input_manager.py

```diff
@@ -107,9 +108,25 @@ def read_parameter_file(self, filename):
         :param filename: the path to the parameter file
         :return: dictionary of parameters read from the parameter file
         """
-        parameter_set_name, extension = os.path.splitext(os.path.basename(filename))
+        # Get File location
+        file_path = Path(filename)
+        parameter_set_name, extension = file_path.stem, file_path.suffix
+        if filename[0] == ".":
+            # All relative need to be in reference to root folder
+            fn_abspart = filename.split("./")[-1]
+            temp_dir = self.root_dir
+            # if parent folder is required, move to parent directory number
+            # of times in relative path (.../ will backup 2 directories)
+            n_parents = filename.count("..")
+            for pcnt in range(n_parents):
+                temp_dir = temp_dir.parent
+            param_file = temp_dir / fn_abspart
+        else:
+            param_file = Path(filename)
+
```

Comment:
Sure. My guess is that running the model with working directory anywhere but where `ldar_sim_main.py` exists will not work because of the relative imports. The solution is probably adding import paths relative to `ldar_sim_main.py` with `sys.path.append (os.path.dirname(__file__))` or something similar - I can't remember the syntax exactly - but I have this suspicion that there are additional places where the working directory matters.

So it looks like relative imports from an arbitrary working directory are likely never going to work.

This looks fine.

Reply:
Actually the root directory gets determined already based on (os.path.dirname(__file__)). So all files that lead will a relative call get directed from there, and All file calls are independent of the working directory for that reason (someone using another working directory). We need to test it on Linux / macOS but it does work with windows. I think you might be right about the working directory being used in other places in the file. I tried to switch all to relative refs to Root, but I may have missed a few.